library/src-3.x/scala/quoted/package.scala

```diff
@@ -3,19 +3,48 @@ package scala
 package object quoted {
 
   /** Evaluate the contents of this expression and return the result.
+   *  It provides a new QuoteContext that is only valid within the scope the argument.
    *
    *  Usage:
    *  ```
-   *  val e: T = run {
-   *   expr
+   *  val e: T = run { // (given qctx: QuoteContext) =>
+   *    expr
    *  }
    *  ```
    *  where `expr: Expr[T]`
-    *
+   *
+   *  This method shoul not be called in a context where there is already has a QuoteContext.
+   *
    *  May throw a FreeVariableError on expressions that came from a macro.
    */
   def run[T](expr: given QuoteContext => Expr[T]) given (toolbox: Toolbox): T = toolbox.run(expr given _)
 
+  /** Provide a new quote context within the scope of the argument that is only valid within the scope the argument.
+   *  Return the result of the argument.
+   *
+   *  Usage:
+   *  ```
+   *  val e: T = withNewQuoteContext { // (given qctx: QuoteContext) =>
+   *    thunk
+   *  }
+   *  ```
+   *  where `thunk: T`
+   *
+   *  This method shoul not be called in a context where there is already has a QuoteContext.
+   */
+  def withNewQuoteContext[T](thunk: given QuoteContext => T) given (toolbox: Toolbox): T = {
```

Comment:
Nitpick: `withQuoteContext` seems to be a better name, as the provided `QuoteContext` is assumed to be scoped by the `thunk` --- there is no old context there, so `new` does not convey any meaning.

Another thing for documentation is: what if programmer nest `run` or `withQuoteContext` --- it's forbidden or allowed.